src/ucp/wireup/address.c

```diff
@@ -506,30 +507,36 @@ ucp_address_unpack_byte_extended(const void *ptr, size_t value_mask,
     return (void*)ptr;
 }
 
+static size_t ucp_address_md_mask(ucp_object_version_t addr_version)
+{
+    return (addr_version == UCP_OBJECT_VERSION_V1) ?
+           UCP_ADDRESS_FLAG_MD_MASK_V1 : UCP_ADDRESS_FLAG_MD_MASK;
+}
+
 static void *
 ucp_address_pack_md_info(void *ptr, int is_empty_dev, uint64_t md_flags,
                          ucp_md_index_t md_index,
                          ucp_object_version_t addr_version)
 {
-    void *flags_ptr = ptr;
-
-    ptr = ucp_address_pack_byte_extended(ptr, md_index,
-                                         UCP_ADDRESS_FLAG_MD_MASK,
-                                         addr_version);
+    uint8_t *flags_ptr = ptr;
+    size_t mask        = ucp_address_md_mask(addr_version);
 
     if (is_empty_dev) {
-        *(uint8_t*)flags_ptr |= UCP_ADDRESS_FLAG_MD_EMPTY_DEV;
+        *flags_ptr |= UCP_ADDRESS_FLAG_MD_EMPTY_DEV;
     }
 
-    if (md_flags & UCT_MD_FLAG_ALLOC) {
-        *(uint8_t*)flags_ptr |= UCP_ADDRESS_FLAG_MD_ALLOC;
-    }
+    if (addr_version == UCP_OBJECT_VERSION_V1) {
+        /* Preserve wire protocol even though these flags are not used */
+        if (md_flags & UCT_MD_FLAG_ALLOC) {
+            *flags_ptr |= UCP_ADDRESS_FLAG_MD_ALLOC;
+        }
 
-    if (md_flags & UCT_MD_FLAG_REG) {
-        *(uint8_t*)flags_ptr |= UCP_ADDRESS_FLAG_MD_REG;
+        if (md_flags & UCT_MD_FLAG_REG) {
+            *flags_ptr |= UCP_ADDRESS_FLAG_MD_REG;
+        }
     }
 
-    return ptr;
+    return ucp_address_pack_byte_extended(ptr, md_index, mask, addr_version);
 }
 
 static void *ucp_address_unpack_md_info(const void *ptr,
```

Comment:
minor:
shall the function return `const void*`, also `ucp_address_unpack_byte_extended()`/`ucp_address_unpack_md_info()` should be updated too (the same way as we have for `ucp_address_unpack_tl_length()`)?